Drivers/imu/inc/icm42688.hpp

```diff
@@ -0,0 +1,109 @@
+// icm42688.hpp
+
+#ifndef ICM42688_HPP
+#define ICM42688_HPP
+
+#include "imu_driver.hpp"
+#include <stdint.h>
+
+class ICM42688 : public IMUDriver {
+    public:
+        /**
```

Comment:
Probably the constructor is missing here. It would be great if we could make the constructor take in an SPI instance and the CS pin. This way, this driver is not limited to only working with spi1 and cs pin at GPIOA4.